src/kcas_data/accumulator.ml

```diff
@@ -0,0 +1,52 @@
+open Kcas
+
+let n_way_max = Domain.recommended_domain_count () |> Bits.ceil_pow_2
+let n_way_default = n_way_max |> Int.min 8
+
+type t = int Loc.t array
+
+let make ?n_way n =
+  let n_way =
+    match n_way with
+    | None -> n_way_default
+    | Some n_way -> n_way |> Int.min n_way_max |> Bits.ceil_pow_2
+  in
+  Array.init n_way (fun i -> Loc.make (if i = 0 then n else 0))
+
+let n_way_of = Array.length
+
+let get_self a =
+  let h = (Domain.self () :> int) in
+  (* TODO: Consider mixing the bits of [h] to get better distribution *)
+  Array.unsafe_get a (h land (Array.length a - 1))
```

Comment:
The problem here is that `Domain.self ()` does not make for a very good hash function and the bitwise AND will just discard any high order bits.  One potential solution would be to use a fast integer hash function here.   Probably a 16 or fewer bit hash function would be good enough for this purpose.